server/routes/addInvolvedStaff.ts

```diff
@@ -134,7 +164,62 @@ export default class AddInvolvedStaffRoutes {
     if (!query) {
       return res.redirect(paths.staffInvolved(bookingId))
     }
-    const name = query.username
-    return res.render('formPages/addingStaff/staff-member-not-found', { name, bookingId })
+    const { firstName, lastName } = query
+    const name = properCaseFullName(`${firstName} ${lastName}`)
+    return res.render('formPages/addingStaff/staff-member-not-found', { bookingId, name })
+  }
+
+  public viewSelectStaffMember = async (req: Request, res: Response): Promise<void> => {
+    const { bookingId } = req.params
+    const query = getFromFlash(req, 'query')
+    if (!query) {
+      return res.redirect(paths.staffInvolved(bookingId))
+    }
+    const { firstName, lastName } = query
+    const staff = await this.draftReportService.findUsers(
+      await this.systemToken(req.user.username),
+      firstName,
+      lastName
+    )
+    return res.render('formPages/addingStaff/select-staff-member', {
+      bookingId,
+      firstName,
+      lastName,
+      staff,
+      errors: req.flash('errors'),
+    })
+  }
+
+  public submitSelectStaffMember = async (req: Request, res: Response): Promise<void> => {
+    const { bookingId } = req.params
+    const { selectedStaffUsername, firstName, lastName } = req.body
+
+    if (!selectedStaffUsername) {
+      const errors = [
+        {
+          text: 'Select the staff member you want to add',
+          href: '#user-1',
+        },
+      ]
+      req.flash('query', { firstName, lastName })
+      req.flash('errors', errors)
+      return res.redirect(paths.selectStaffMember(bookingId))
+    }
+
+    const result = await this.draftReportService.addDraftStaffByUsername(
+      res.locals.user,
+      parseInt(bookingId, 10),
+      selectedStaffUsername
+    )
+
+    switch (result) {
+      case AddStaffResult.ALREADY_EXISTS:
+      case AddStaffResult.SUCCESS:
+      case AddStaffResult.SUCCESS_UNVERIFIED: {
```

Comment:
be forgiving if user happened to already be added - like in a separate tab or something